From 55df611bf0440c692f539f5c3bd09112a9cdc247 Mon Sep 17 00:00:00 2001
From: Richard Zhu <hongxing.zhu@nxp.com>
Date: Thu, 21 Sep 2017 16:46:09 +0800
Subject: [PATCH 2609/5242] MLK-16530-2 clk: imx8qm: add the cm40 ipg clk

commit  558204d6010125079ba4ecba93001656399ddd6c from
https://source.codeaurora.org/external/imx/linux-imx.git

Add the cm40 ipg clk
BuildInfo: SCFW 9e9f6ec6, IMX-MKIMAGE 0, ATF 0

Signed-off-by: Richard Zhu <hongxing.zhu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/clk/imx/clk-imx8qm.c             |    1 +
 include/dt-bindings/clock/imx8qm-clock.h |    5 ++++-
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/clk/imx/clk-imx8qm.c b/drivers/clk/imx/clk-imx8qm.c
index ba1a143..29dcdd0 100644
--- a/drivers/clk/imx/clk-imx8qm.c
+++ b/drivers/clk/imx/clk-imx8qm.c
@@ -577,6 +577,7 @@ static int imx8qm_clk_probe(struct platform_device *pdev)
 	clks[IMX8QM_HDMI_RX_SAI0_RX_BCLK]  = imx_clk_fixed("hdmi_rx_sai0_rx_bclk", 0);
 	clks[IMX8QM_SAI6_RX_BCLK]          = imx_clk_fixed("sai6_rx_bclk", 0);
 	clks[IMX8QM_HDMI_TX_SAI0_TX_BCLK]  = imx_clk_fixed("hdmi_tx_sai0_tx_bclk", 0);
+	clks[IMX8QM_CM40_IPG_CLK]	= imx_clk_fixed("ipg_cm40_clk_root", SC_132MHZ);
 
 	np_acm = of_find_compatible_node(NULL, NULL, "nxp,imx8qm-acm");
 	if (np_acm) {
diff --git a/include/dt-bindings/clock/imx8qm-clock.h b/include/dt-bindings/clock/imx8qm-clock.h
index dcb6c2c..a5b57c4 100644
--- a/include/dt-bindings/clock/imx8qm-clock.h
+++ b/include/dt-bindings/clock/imx8qm-clock.h
@@ -823,6 +823,9 @@
 #define IMX8QM_DC1_DISP0_SEL				770
 #define IMX8QM_DC1_DISP1_SEL				771
 
-#define IMX8QM_CLK_END					772
+/* CM40 */
+#define IMX8QM_CM40_IPG_CLK				772
+
+#define IMX8QM_CLK_END					773
 
 #endif /* __DT_BINDINGS_CLOCK_IMX8QM_H */
-- 
1.7.9.5

